### PR TITLE
Add MicroForwarder

### PR DESCRIPTION
**Pull request by jefft0**
A MicroForwarder is an in-memory object with a PIT and a FIB and functionality similar to a full NDN forwarder. The application can create a Face to the MicroForwarder instead of a normal forwarder. (This is useful on systems that don't support NFD.) The application can create forwarder faces and routes.

This pull request has three commits. The first commit adds the `ElementListenerLite` class which exposes existing functionality in the public API so that it can be used by the MicroForwarder.

The second commit adds the `MicroForwarder` to the ndn-ind-tools library (which is always compiled and installed along with the main ndn-ind library).  It also adds the class `MicroForwarderTransport`. When the application creates a Face, it uses this to connect to the MicroForwarder.

The third commit updates the VisualStudio project to add the MicroForwarder files.

### CODE REVIEW

**Review by OperantSWDev — DISMISSED** [review]
This is super cool that you were able to leverage your previous work on the microforwarder.  I stared at this section for a lot of hours during the FleetLink development!
 
_// Now process as Interest or Data._

Definitely something we can use a lot going forward I hope.

File: include/ndn-ind-tools/micro-forwarder/micro-forwarder-transport.hpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/**
+ * Copyright (C) 2021 Operant Networks, Incorporated.
+ * @author: Jeff Thompson <jefft0@gmail.com>
```
Comment (maysax) [inline include/ndn-ind-tools/micro-forwarder/micro-forwarder-transport.hpp:4]: pl. remove the email address

File: tools/micro-forwarder/micro-forwarder-transport.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/**
+ * Copyright (C) 2021 Operant Networks, Incorporated.
+ * @author: Jeff Thompson <jefft0@gmail.com>
```
Comment (maysax) [inline tools/micro-forwarder/micro-forwarder-transport.cpp:4]: same, pl. remove email address

File: tools/micro-forwarder/micro-forwarder.cpp [hunk 3]
```diff
@@ -0,0 +1,345 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/**
+ * Copyright (C) 2021 Operant Networks, Incorporated.
+ * @author: Jeff Thompson <jefft0@gmail.com>
```
Comment (maysax) [inline tools/micro-forwarder/micro-forwarder.cpp:4]: same

File: tools/micro-forwarder/micro-forwarder-transport.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+MicroForwarderTransport::isLocal(const Transport::ConnectionInfo& connectionInfo)
+{
+  return true;
+}
+
+bool
+MicroForwarderTransport::isAsync() { return false; }
+
+void
+MicroForwarderTransport::connect
```
Comment (maysax) [inline tools/micro-forwarder/micro-forwarder-transport.cpp:55]: Like in previous files can we pl. add the description comments before each function in this file as well. Thanks

**Review by maysax — APPROVED** [review]
Approved, contingent on addressing the mentioned minor comments

File: include/ndn-ind/lite/encoding/element-reader-lite.hpp [hunk 5]
```diff
@@ -0,0 +1,81 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/**
+ * Copyright (C) 2021 Operant Networks, Incorporated.
+ * @author: Jeff Thompson <jefft0@gmail.com>
```
Comment (maysax) [inline include/ndn-ind/lite/encoding/element-reader-lite.hpp:4]: same

File: src/lite/encoding/element-reader-lite.cpp [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/**
+ * Copyright (C) 2021 Operant Networks, Incorporated.
+ * @author: Jeff Thompson <jefft0@gmail.com>
```
Comment (maysax) [inline src/lite/encoding/element-reader-lite.cpp:4]: same

**Review by maysax — APPROVED** [review]
Approve, contingent on addressing minor comments.

File: tools/micro-forwarder/micro-forwarder-transport.cpp (hunk 4 above)
Comment (jefft0) [reply tools/micro-forwarder/micro-forwarder-transport.cpp:55]: The comments go in the header file. Is that OK? https://github.com/operantnetworks/ndn-ind/blob/MicroForwarder/include/ndn-ind-tools/micro-forwarder/micro-forwarder-transport.hpp#L92-L99

**Review by KLIU-opnt — DISMISSED** [review]

File: include/ndn-ind-tools/micro-forwarder/micro-forwarder-transport.hpp (hunk 1 above)
Comment (jefft0) [reply include/ndn-ind-tools/micro-forwarder/micro-forwarder-transport.hpp]: I pushed changes to remove the email address.

File: tools/micro-forwarder/micro-forwarder-transport.cpp (hunk 2 above)
Comment (jefft0) [reply tools/micro-forwarder/micro-forwarder-transport.cpp]: I pushed changes to remove the email address.

File: tools/micro-forwarder/micro-forwarder.cpp (hunk 3 above)
Comment (jefft0) [reply tools/micro-forwarder/micro-forwarder.cpp]: I pushed changes to remove the email address.

File: include/ndn-ind/lite/encoding/element-reader-lite.hpp (hunk 5 above)
Comment (jefft0) [reply include/ndn-ind/lite/encoding/element-reader-lite.hpp]: I pushed changes to remove the email address.

File: src/lite/encoding/element-reader-lite.cpp (hunk 6 above)
Comment (jefft0) [reply src/lite/encoding/element-reader-lite.cpp]: I pushed changes to remove the email address.

**Review by KLIU-opnt — APPROVED** [review]

**Review by maysax — APPROVED** [review]

**Pull request merged**